File: instalock.py
import pyautogui as pg
import time

class Instalock:
    def __init__(self) -> None:
        pass

    def locker(self, x, y):
        for i in range(50):
            pg.moveTo(x,y)
            pg.click()
            pg.moveTo(950, 790)
            pg.click()

    def lock_it(self, data):

        data = data.lower()
        topList = ["astra", "breach", "brim", "chamber", "cypher", "fade", "gekko", "harbor", "jett", "kayo", "killjoy"]
        bottomList = ["neon", "omen", "pho", "raze", "reyna", "sage", "skye", "sova", "viper", "yoru"]
        referance = [560, 935]
        time.sleep(3)
        counter = 0
        for i in topList:
            if data == i:
                print(f"locker coords: {referance[0] + counter * 80}, {referance[1]}")
                self.locker(referance[0] + counter * 80, referance[1])
                
            counter += 1
        counter = 0
        for i in bottomList:
            if data == i:
                print(f"locker coords: {referance[0] + counter * 80}, {referance[1] + 80}")
                self.locker(referance[0] + counter * 80, referance[1] + 80)
            
            counter += 1

    def log(self):
        while True:
            time.sleep(2)
            print(pg.position())

#log()
banner = """  menu:
    astra, breach, brim, chamber, cypher, fade, gekko, harbor, jett, kayo, killjoy
    neon, omen, pho, raze, reyna, sage, skye, sova, viper, yoru
    """
print(banner)

i = Instalock()
i.lock_it("jett")

File: instalock.c
#include <windows.h>
#include <stdio.h>
#include <unistd.h>
//c version
void click(){
	mouse_event(MOUSEEVENTF_LEFTDOWN, 0, 0, 0, 0);
    mouse_event(MOUSEEVENTF_LEFTUP, 0, 0, 0, 0);
}

void locker(int x, int y){
	for(int i = 0; i<500; i ++){
		SetCursorPos(x, y);
		click();
		SetCursorPos(950, 790);
		click();
	}
}

void lock_it(char* data) {
    char* topList[11] = {"astra", "breach", "brim", "chamber", "cypher", "fade", "gekko", "harbor", "jett", "kayo", "killjoy"};
    char* bottomList[10] = {"neon", "omen", "pho", "raze", "reyna", "sage", "skye", "sova", "viper", "yoru"};
    int referance[2] = {560, 935};
    int counter = 0;
    int i;
    for (i = 0; i < 11; i++) {
        if (strcmp(data, topList[i]) == 0) {
            printf("locker coords: %d, %d\n", referance[0] + counter * 80, referance[1]);
            locker(referance[0] + counter * 80, referance[1]);
        }
        counter++;
    }
    counter = 0;
    for (i = 0; i < 10; i++) {
        if (strcmp(data, bottomList[i]) == 0) {
            printf("locker coords: %d, %d\n", referance[0] + counter * 80, referance[1] + 80);
            locker(referance[0] + counter * 80, referance[1] + 80);
        }
        counter++;
    }
}
int main()
{	
	sleep(3);
	lock_it("jett");
    return 0;
}

